code/src/java/pcgen/gui2/facade/CharacterFacadeImpl.java

```diff
@@ -2188,6 +2215,18 @@ void refreshLanguageList()
 		{
 			todoManager.removeTodo("in_sumTodoBonusLanguage");
 			todoManager.removeTodo("in_sumTodoBonusLanguageFirstOnly");
+
+			/* Ensure the bonus language chooser is removed, if it exists.*/
+			Iterator<LanguageChooserFacade> itr = langChoosersList.iterator();
+			while (itr.hasNext())
```

Comment:
Let's use a enhanced forloop here?

Reply:
The only reason I didn't is because using an iterator allows for removing the choosers in-place. I don't know of a way to do this with an enhanced for loop. I could use a normal for loop with the iterator, but it would be of the form:

```java
for (LanguageChooserFacade chooser = itr.next();  itr.hasNext();)
{
	/* If we find an add bonus chooser, remove it.*/
	if (chooser.getName().equals(LanguageBundle.getString("in_sumLangBonus")))
	{
		itr.remove();
	}
}
```

Is that reasonable?

Reply:
Works for me other than the existing comment about using the Key or other non translated attribute